application/src/main/java/org/togetherjava/tjbot/commands/bytecode/BytecodeCommand.java

```diff
@@ -0,0 +1,269 @@
+package org.togetherjava.tjbot.commands.bytecode;
+
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.TextChannel;
+import net.dv8tion.jda.api.events.interaction.command.MessageContextInteractionEvent;
+import net.dv8tion.jda.api.events.message.MessageDeleteEvent;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.events.message.MessageUpdateEvent;
+import net.dv8tion.jda.api.interactions.InteractionHook;
+import net.dv8tion.jda.api.interactions.commands.build.Commands;
+import org.jetbrains.annotations.NotNull;
+import org.togetherjava.tjbot.commands.BotCommandAdapter;
+import org.togetherjava.tjbot.commands.CommandVisibility;
+import org.togetherjava.tjbot.commands.MessageContextCommand;
+import org.togetherjava.tjbot.commands.MessageReceiver;
+import org.togetherjava.tjbot.imc.CompilationResult;
+import org.togetherjava.tjbot.imc.CompileInfo;
+import org.togetherjava.tjbot.imc.InMemoryCompiler;
+import org.togetherjava.tjbot.imc.JavacOption;
+import org.togetherjava.tjbot.javap.Javap;
+import org.togetherjava.tjbot.javap.JavapOption;
+
+import javax.tools.Diagnostic;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * Bytecode command that uses the message context commands Example usage:
+ * <p>
+ *
+ * <pre>
+ * {@code
+ *     ```java
+ *     class Test {
+ *         public void foo() {
+ *             System.out.println("Hello World!");
+ *         }
+ *     }
+ *     ```
+ * }
+ * </pre>
+ *
+ * then apply the application
+ */
+public final class BytecodeCommand extends BotCommandAdapter
+        implements MessageContextCommand, MessageReceiver {
+    private static final String CODE_BLOCK_OPENING = "```\n";
+    private static final String CODE_BLOCK_CLOSING = "\n```";
+
+    private final Pattern codeBlockExtractorPattern =
+            Pattern.compile("```(?:java)?\\s*([\\w\\W]+)```|``?([\\w\\W]+)``?");
```

Comment:
this regex is way too complex to merge it without any throughout explanation whatsoever. nobody is able to understand this or maintain it. u probably also already forgot what it does and have to re-analyze it in ur head first.

please add some in-depth explanation of how this works, thanks.

Reply:
what exactly do you want explained? to me that regex looks very understandable if you actually try reading it

Reply:
> if you actually try reading it

Your requirements for "reading" are extremely high.

Not everyone is fine wasting 20m of their time, assuming they have good Regex knowledge.

What has to be explained? What your pattern does has to be explained.
What do the groups stand for? What does `[\\w\\W]+` do?

I don't mean an essay

I just mean

> It looks for codeblocks, here it checks or it's a java code block, or undecided.
> This means that PHP codeblocks and such get ignored

etc

Reply:
also u can give a quick outline which group id contains what.

Reply:
there's only one group id